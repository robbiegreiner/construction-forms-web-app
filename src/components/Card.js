import React, { Component } from 'react';
import '../styles/Card.css';

class Card extends Component {
  render() {
    return (
      <div className={ this.props.formDetails ? "form-card" : "card" }>
        <h5>{this.props.header}</h5>
        <p>{this.props.body}</p>
        {
          this.props.projectInfo &&
            <div>
              <p>Public: {this.props.projectInfo[0].toString()}</p>
              <p>Union: {this.props.projectInfo[1].toString()}</p>
            </div>
        }
        {
          this.props.employeeInfo &&
            <div>
              <p>Phone: {this.props.employeeInfo[0]}</p>
              <p>Email: {this.props.employeeInfo[1]}</p>
            </div>
        }
        {
          this.props.formDetails && 
            <div className="form-card">
              <p>Employee Email: {this.props.formDetails[0]}</p>
              <p>Project ID: {this.props.formDetails[1]}</p>
              <p>Company: {this.props.formDetails[2]}</p>
              <p>Date: {this.props.formDetails[3]}</p>
              <p>Firewatch Requirement: {this.props.formDetails[4]}</p>
              <p>Area Inspected: {this.props.formDetails[5] ? 'Yes' : 'No'}</p>
              <p>Fire Extinguisher: {this.props.formDetails[6] ? 'Yes' : 'No'}</p>
              <p>Flammables Removed: {this.props.formDetails[7]? 'Yes' : 'No'}</p>
              <p>Sprinkler Heads Protected: {this.props.formDetails[8] ? 'Yes' : 'No'}</p>
              <img src={this.props.formDetails[9]}/>
            </div>
        }
      </div>
    )
  }
}

export default Card;
