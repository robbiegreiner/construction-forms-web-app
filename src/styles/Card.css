.form-card {
  background-color: green;
  background-color: #f2f3f7;
  padding: 10px;
  margin: 10px;
  border-radius: 4px;
}

.form-card h5 {
  font-size: 16px;
}

.form-card p {
  font-size: 12px;
}

.form-card img {
  width: 66%;
  height: 33%;
}
