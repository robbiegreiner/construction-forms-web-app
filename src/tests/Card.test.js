/* eslint-disable no-unused-vars, no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import Card from '../components/Card';

describe('Card tests', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Card />);
  });

  it('should render without crashing', () => {
    shallow(<Card />);
  });

  it('should have a class of card by default', () => {
    expect(wrapper.hasClass('card')).toEqual(true);
  });

  it('should have a class of form-card when passed formDetails', () => {
    wrapper = shallow(<Card formDetails={[]} />);
    expect(wrapper.hasClass('form-card')).toEqual(true);
  });

  it('should render as a project card when passed projectInfo', () => {
    const mockProjectInfo = [true, true];

    wrapper = shallow(<Card 
      header='The Alan Parsons Project'
      body='Sirius'
      projectInfo={ mockProjectInfo }
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render as an employee card when passed employeeInfo', () => {
    const mockEmployeeInfo = ['555-555-5556', 'alanparsons@project.com'];

    wrapper = shallow(<Card 
      header='Alan Parson'
      body='Musician'
      employeeInfo={ mockEmployeeInfo }
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render as a form card when passed formInfo', () => {
    const mockFormDetails = [
      'alanparsons@project.com',
      '12',
      'The APP Inc',
      '2018-01-08',
      '1 hour',
      true,
      true,
      true,
      true,
      'xxx'
    ];

    wrapper = shallow(<Card 
      header='Hotwork Permit - Alan Parson'
      formDetails={ mockFormDetails }
    />);

    expect(wrapper).toMatchSnapshot();
  });


  it('should match the snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
