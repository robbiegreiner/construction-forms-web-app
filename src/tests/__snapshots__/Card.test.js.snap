// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests should match the snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5 />,
        <p
          className={null}
/>,
        undefined,
        undefined,
        undefined,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": null,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      undefined,
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5 />,
          <p
            className={null}
/>,
          undefined,
          undefined,
          undefined,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": null,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        undefined,
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card tests should render as a form card when passed formInfo 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    formDetails={
        Array [
            "alanparsons@project.com",
            "12",
            "The APP Inc",
            "2018-01-08",
            "1 hour",
            true,
            true,
            true,
            true,
            "xxx",
          ]
    }
    header="Hotwork Permit - Alan Parson"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Hotwork Permit - Alan Parson
</h5>,
        <p
          className={null}
/>,
        undefined,
        undefined,
        <div
          className="form-card"
>
          <p>
                    Employee Email: 
                    alanparsons@project.com
          </p>
          <p>
                    Project ID: 
                    12
          </p>
          <p>
                    Company: 
                    The APP Inc
          </p>
          <p>
                    Date: 
                    2018-01-08
          </p>
          <p>
                    Firewatch Requirement: 
                    1 hour
          </p>
          <p>
                    Area Inspected: 
                    Yes
          </p>
          <p>
                    Fire Extinguisher: 
                    Yes
          </p>
          <p>
                    Flammables Removed: 
                    Yes
          </p>
          <p>
                    Sprinkler Heads Protected: 
                    Yes
          </p>
          <img
                    alt="Hotwork Permit - Alan Parson Signature"
                    src="xxx"
          />
</div>,
      ],
      "className": "form-card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hotwork Permit - Alan Parson",
        },
        "ref": null,
        "rendered": "Hotwork Permit - Alan Parson",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": null,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      undefined,
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Employee Email: 
              alanparsons@project.com
</p>,
            <p>
              Project ID: 
              12
</p>,
            <p>
              Company: 
              The APP Inc
</p>,
            <p>
              Date: 
              2018-01-08
</p>,
            <p>
              Firewatch Requirement: 
              1 hour
</p>,
            <p>
              Area Inspected: 
              Yes
</p>,
            <p>
              Fire Extinguisher: 
              Yes
</p>,
            <p>
              Flammables Removed: 
              Yes
</p>,
            <p>
              Sprinkler Heads Protected: 
              Yes
</p>,
            <img
              alt="Hotwork Permit - Alan Parson Signature"
              src="xxx"
/>,
          ],
          "className": "form-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Employee Email: ",
                "alanparsons@project.com",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Employee Email: ",
              "alanparsons@project.com",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Project ID: ",
                "12",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Project ID: ",
              "12",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Company: ",
                "The APP Inc",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Company: ",
              "The APP Inc",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Date: ",
                "2018-01-08",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Date: ",
              "2018-01-08",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Firewatch Requirement: ",
                "1 hour",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Firewatch Requirement: ",
              "1 hour",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Area Inspected: ",
                "Yes",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Area Inspected: ",
              "Yes",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Fire Extinguisher: ",
                "Yes",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Fire Extinguisher: ",
              "Yes",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Flammables Removed: ",
                "Yes",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Flammables Removed: ",
              "Yes",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Sprinkler Heads Protected: ",
                "Yes",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Sprinkler Heads Protected: ",
              "Yes",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Hotwork Permit - Alan Parson Signature",
              "src": "xxx",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Hotwork Permit - Alan Parson
</h5>,
          <p
            className={null}
/>,
          undefined,
          undefined,
          <div
            className="form-card"
>
            <p>
                        Employee Email: 
                        alanparsons@project.com
            </p>
            <p>
                        Project ID: 
                        12
            </p>
            <p>
                        Company: 
                        The APP Inc
            </p>
            <p>
                        Date: 
                        2018-01-08
            </p>
            <p>
                        Firewatch Requirement: 
                        1 hour
            </p>
            <p>
                        Area Inspected: 
                        Yes
            </p>
            <p>
                        Fire Extinguisher: 
                        Yes
            </p>
            <p>
                        Flammables Removed: 
                        Yes
            </p>
            <p>
                        Sprinkler Heads Protected: 
                        Yes
            </p>
            <img
                        alt="Hotwork Permit - Alan Parson Signature"
                        src="xxx"
            />
</div>,
        ],
        "className": "form-card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hotwork Permit - Alan Parson",
          },
          "ref": null,
          "rendered": "Hotwork Permit - Alan Parson",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": null,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        undefined,
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Employee Email: 
                alanparsons@project.com
</p>,
              <p>
                Project ID: 
                12
</p>,
              <p>
                Company: 
                The APP Inc
</p>,
              <p>
                Date: 
                2018-01-08
</p>,
              <p>
                Firewatch Requirement: 
                1 hour
</p>,
              <p>
                Area Inspected: 
                Yes
</p>,
              <p>
                Fire Extinguisher: 
                Yes
</p>,
              <p>
                Flammables Removed: 
                Yes
</p>,
              <p>
                Sprinkler Heads Protected: 
                Yes
</p>,
              <img
                alt="Hotwork Permit - Alan Parson Signature"
                src="xxx"
/>,
            ],
            "className": "form-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Employee Email: ",
                  "alanparsons@project.com",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Employee Email: ",
                "alanparsons@project.com",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Project ID: ",
                  "12",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Project ID: ",
                "12",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Company: ",
                  "The APP Inc",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Company: ",
                "The APP Inc",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Date: ",
                  "2018-01-08",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Date: ",
                "2018-01-08",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Firewatch Requirement: ",
                  "1 hour",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Firewatch Requirement: ",
                "1 hour",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Area Inspected: ",
                  "Yes",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Area Inspected: ",
                "Yes",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Fire Extinguisher: ",
                  "Yes",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Fire Extinguisher: ",
                "Yes",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Flammables Removed: ",
                  "Yes",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Flammables Removed: ",
                "Yes",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Sprinkler Heads Protected: ",
                  "Yes",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Sprinkler Heads Protected: ",
                "Yes",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Hotwork Permit - Alan Parson Signature",
                "src": "xxx",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card tests should render as a project card when passed projectInfo 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    body="Sirius"
    header="The Alan Parsons Project"
    projectInfo={
        Array [
            true,
            true,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          The Alan Parsons Project
</h5>,
        <p
          className={null}
>
          Sirius
</p>,
        <div>
          <p>
                    Public: 
                    Yes
          </p>
          <p>
                    Union: 
                    Yes
          </p>
</div>,
        undefined,
        undefined,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Alan Parsons Project",
        },
        "ref": null,
        "rendered": "The Alan Parsons Project",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sirius",
          "className": null,
        },
        "ref": null,
        "rendered": "Sirius",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Public: 
              Yes
</p>,
            <p>
              Union: 
              Yes
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Public: ",
                "Yes",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Public: ",
              "Yes",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Union: ",
                "Yes",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Union: ",
              "Yes",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            The Alan Parsons Project
</h5>,
          <p
            className={null}
>
            Sirius
</p>,
          <div>
            <p>
                        Public: 
                        Yes
            </p>
            <p>
                        Union: 
                        Yes
            </p>
</div>,
          undefined,
          undefined,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Alan Parsons Project",
          },
          "ref": null,
          "rendered": "The Alan Parsons Project",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sirius",
            "className": null,
          },
          "ref": null,
          "rendered": "Sirius",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Public: 
                Yes
</p>,
              <p>
                Union: 
                Yes
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Public: ",
                  "Yes",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Public: ",
                "Yes",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Union: ",
                  "Yes",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Union: ",
                "Yes",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card tests should render as an employee card when passed employeeInfo 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    body="Musician"
    employeeInfo={
        Array [
            "555-555-5556",
            "alanparsons@project.com",
          ]
    }
    header="Alan Parson"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Alan Parson
</h5>,
        <p
          className={null}
>
          Musician
</p>,
        undefined,
        <div>
          <p>
                    555-555-5556
          </p>
          <p>
                    alanparsons@project.com
          </p>
</div>,
        undefined,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Alan Parson",
        },
        "ref": null,
        "rendered": "Alan Parson",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Musician",
          "className": null,
        },
        "ref": null,
        "rendered": "Musician",
        "type": "p",
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              555-555-5556
</p>,
            <p>
              alanparsons@project.com
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "555-555-5556",
            },
            "ref": null,
            "rendered": "555-555-5556",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "alanparsons@project.com",
            },
            "ref": null,
            "rendered": "alanparsons@project.com",
            "type": "p",
          },
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Alan Parson
</h5>,
          <p
            className={null}
>
            Musician
</p>,
          undefined,
          <div>
            <p>
                        555-555-5556
            </p>
            <p>
                        alanparsons@project.com
            </p>
</div>,
          undefined,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Alan Parson",
          },
          "ref": null,
          "rendered": "Alan Parson",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Musician",
            "className": null,
          },
          "ref": null,
          "rendered": "Musician",
          "type": "p",
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                555-555-5556
</p>,
              <p>
                alanparsons@project.com
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "555-555-5556",
              },
              "ref": null,
              "rendered": "555-555-5556",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "alanparsons@project.com",
              },
              "ref": null,
              "rendered": "alanparsons@project.com",
              "type": "p",
            },
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
